atom/browser/api/atom_api_global_shortcut.cc

```diff
@@ -38,6 +39,22 @@ void GlobalShortcut::OnKeyPressed(const ui::Accelerator& accelerator) {
   accelerator_callback_map_[accelerator].Run();
 }
 
+bool GlobalShortcut::RegisterAll(
+    const std::vector<ui::Accelerator>& accelerators,
+    const base::Closure& callback) {
+  for (auto& accelerator : accelerators) {
+    GlobalShortcutListener* listener = GlobalShortcutListener::GetInstance();
+    if (!listener->RegisterAccelerator(accelerator, this)) {
+      // unregister all shortcuts if any failed
+      UnregisterAll();
```

Comment:
This introduces another issue where if I have code that looks like this.

```js
register('Control+A')
registerAll(['Control+B', 'BadThing'])
```

just because the `registerAll` call fails we shouldn't unregister the `Control + A` handler which is what this method would do